src/Mod/Assembly/Gui/ViewProviderAssembly.cpp

```diff
@@ -700,6 +684,45 @@ bool ViewProviderAssembly::getSelectedObjectsWithinAssembly(bool addPreselection
     return !docsToMove.empty();
 }
 
+void ViewProviderAssembly::collectMovableObjects(App::DocumentObject* selRoot,
+                                                 std::string& subNamePrefix,
```

Comment:
I notice that you are passing this by a non-const reference -- did you intend for the code below to update it when `newSubNamePrefix` is created? Otherwise it seems that it should be const (though you'll have to adjust `MovingObject` to accommodate that change, it's taking it in as a non-const reference, but then immediately copying it).

Reply:
Yes I used non const because I didn't want to change MovingObject. But really it should be const I think